drop-storage/src/lib.rs

```diff
@@ -32,9 +32,50 @@ pub struct Storage {
 
 const MIGRATIONS_DIR: Dir = include_dir!("$CARGO_MANIFEST_DIR/migrations");
 
+#[cfg(unix)]
+fn prepare_sqlite_file(path: &str) -> io::Result<OpenFlags> {
+    use std::os::unix::prelude::{OpenOptionsExt, PermissionsExt};
+
+    const MODE: u32 = 0o600;
+    let mut flags = OpenFlags::default();
+
+    if path == ":memory:" {
+        return Ok(flags);
+    }
+    // disalbe file creation since we are going to do it ourselves
+    flags &= !OpenFlags::SQLITE_OPEN_CREATE;
+
+    match std::fs::metadata(path) {
+        Ok(meta) => {
+            let mut perm = meta.permissions();
+            if perm.mode() != MODE {
+                perm.set_mode(MODE);
+                std::fs::set_permissions(path, perm)?;
+            }
+        }
+        Err(err) if err.kind() == io::ErrorKind::NotFound => {
+            // create new one
+            std::fs::OpenOptions::new()
+                .write(true)
+                .create_new(true)
+                .mode(MODE)
+                .open(path)?;
+        }
+        Err(err) => return Err(err),
+    }
+
+    Ok(flags)
+}
+
+#[cfg(not(unix))]
+fn prepare_sqlite_file(path: &str) -> io::Result<OpenFlags> {
+    Ok(OpenFlags::default())
+}
```

Comment:
Windows should have teir own 0600 implementation. I think we could even pass it directly and it should just work

Reply:
Sadly it doesn't. At least in the Rust standard library

Reply:
Let me research this :+1:

Reply:
So, as mentioned in today's sync, we leaving the Windows DB permissions for now